tests/projects/backend/test_local.py

```diff
@@ -4,67 +4,75 @@
 from mlflow.projects.backend.local import _get_docker_artifact_storage_cmd_and_envs
 
 
-def test_docker_s3_artifact_cmd_and_envs_from_env():
+def test_docker_s3_artifact_cmd_and_envs_from_env(monkeypatch):
     mock_env = {
         "AWS_SECRET_ACCESS_KEY": "mock_secret",
         "AWS_ACCESS_KEY_ID": "mock_access_key",
         "MLFLOW_S3_ENDPOINT_URL": "mock_endpoint",
         "MLFLOW_S3_IGNORE_TLS": "false",
     }
-    with mock.patch.dict("os.environ", mock_env), mock.patch(
-        "posixpath.exists", return_value=False
-    ):
+    monkeypatch.setenvs(mock_env)
+    with mock.patch("posixpath.exists", return_value=False):
         cmds, envs = _get_docker_artifact_storage_cmd_and_envs("s3://mock_bucket")
         assert cmds == []
         assert envs == mock_env
 
 
-def test_docker_s3_artifact_cmd_and_envs_from_home():
+def test_docker_s3_artifact_cmd_and_envs_from_home(monkeypatch):
     mock_env = {}
-    with mock.patch.dict("os.environ", mock_env), mock.patch(
-        "posixpath.exists", return_value=True
-    ), mock.patch("posixpath.expanduser", return_value="mock_volume"):
+    monkeypatch.setenvs(mock_env)
+    with mock.patch("posixpath.exists", return_value=True), mock.patch(
+        "posixpath.expanduser", return_value="mock_volume"
+    ):
         cmds, envs = _get_docker_artifact_storage_cmd_and_envs("s3://mock_bucket")
         assert cmds == ["-v", "mock_volume:/.aws"]
         assert envs == mock_env
 
 
-def test_docker_wasbs_artifact_cmd_and_envs_from_home():
+def test_docker_wasbs_artifact_cmd_and_envs_from_home(monkeypatch):
     mock_env = {
         "AZURE_STORAGE_CONNECTION_STRING": "mock_connection_string",
         "AZURE_STORAGE_ACCESS_KEY": "mock_access_key",
     }
     wasbs_uri = "wasbs://container@account.blob.core.windows.net/some/path"
-    with mock.patch.dict("os.environ", mock_env), mock.patch(
-        "azure.storage.blob.BlobServiceClient"
-    ):
+    monkeypatch.setenvs(mock_env)
+    with mock.patch("azure.storage.blob.BlobServiceClient"):
         cmds, envs = _get_docker_artifact_storage_cmd_and_envs(wasbs_uri)
         assert cmds == []
         assert envs == mock_env
 
 
-def test_docker_gcs_artifact_cmd_and_envs_from_home():
+def test_docker_gcs_artifact_cmd_and_envs_from_home(monkeypatch):
     mock_env = {
         "GOOGLE_APPLICATION_CREDENTIALS": "mock_credentials_path",
     }
     gs_uri = "gs://mock_bucket"
-    with mock.patch.dict("os.environ", mock_env, clear=True):
-        cmds, envs = _get_docker_artifact_storage_cmd_and_envs(gs_uri)
-        assert cmds == ["-v", "mock_credentials_path:/.gcs"]
-        assert envs == {"GOOGLE_APPLICATION_CREDENTIALS": "/.gcs"}
+    monkeypatch.delenvs(["GOOGLE_APPLICATION_CREDENTIALS"], raising=False)
```

Comment:
@gabrielfu The same thing of unnecessary `delenvs` applied here?

Reply:
that's right :)